drivers/hmis/spec/system/hmis/form_builder_spec.rb

```diff
@@ -0,0 +1,257 @@
+        find("button[aria-label='assessment_date move up']").trigger('click')
+        find("button[type='submit']").trigger('click')
+        assert_no_text 'Save Draft' # Wait for save to complete
+
+        draft.reload
+        expect(draft.definition.dig('item', 0, 'text')).to eq('Assessment Date')
+        expect(draft.definition.dig('item', 1, 'item').size).to eq(1)
+
+        # Test move down out of group
+        find("button[aria-label='custom_question_1 move down']").trigger('click')
+        find("button[type='submit']").trigger('click')
+        assert_no_text 'Save Draft' # Wait for save to complete
+
+        draft.reload
+        expect(draft.definition.dig('item', 0, 'text')).to eq('Assessment Date')
+        expect(draft.definition.dig('item', 1, 'item')).to be_nil
+        expect(draft.definition.dig('item', 2, 'text')).to eq('Custom question 1')
+
+        # Test move down into group
+        find("button[aria-label='assessment_date move down']").trigger('click')
+        find("button[type='submit']").trigger('click')
+        assert_no_text 'Save Draft' # Wait for save to complete
+
+        draft.reload
+        expect(draft.definition.dig('item', 0, 'text')).to eq('Test Custom Assessment')
+        expect(draft.definition.dig('item', 0, 'item').size).to eq(1)
+        expect(draft.definition.dig('item', 0, 'item', 0, 'text')).to eq('Assessment Date')
+
+        # Test move up into group
+        find("button[aria-label='custom_question_1 move up']").trigger('click')
+        find("button[type='submit']").trigger('click')
+        assert_no_text 'Save Draft' # Wait for save to complete
+
+        draft.reload
+        expect(draft.definition.dig('item', 0, 'text')).to eq('Test Custom Assessment')
+        expect(draft.definition.dig('item', 0, 'item').size).to eq(2)
+        expect(draft.definition.dig('item', 0, 'item', 1, 'text')).to eq('Custom question 1')
+      end
+
+      it 'deletes item' do
+        assert_text 'Group: Test Custom Assessment'
+        find("div[aria-label='item section_1']").trigger('click')
+        assert_text 'Date: Assessment Date'
+
+        find("button[aria-label='custom_question_1 item actions']").trigger('click')
+        find('li', text: 'Delete', class: 'MuiMenuItem-root').trigger('click')
+        assert_no_text 'Custom question 1'
+
+        find("button[type='submit']").trigger('click')
+
+        draft.reload
+        expect(draft.definition.dig('item', 0, 'item').size).to eq(1)
+      end
+
+      it 'publishes form' do
+        assert_text 'Group: Test Custom Assessment'
+        click_button 'Preview / Publish'
+        assert_text 'PREVIEWING DRAFT System Test 2'
+        click_button 'Publish'
+        assert_text 'Are you sure you want to publish this form?'
+        click_button 'Confirm'
+        assert_text 'SELECTED FORM System Test 2'
+        assert_text 'Status Published'
+        draft.reload
+        expect(draft.status).to eq('published')
```

Comment:
Neater way is to expect `.to change(draft, :status).to('published')`

Reply:
This works with `reload`: `change { draft.reload.status }.to('published')`